src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Catalog/PackageHasSignatureValidator.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.Exceptions;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.Catalog
+{
+    /// <summary>
+    /// Validates that the package is signed by verifying the presence of the "package signature file"
+    /// in the nupkg. See: https://github.com/NuGet/Home/wiki/Package-Signatures-Technical-Details#-the-package-signature-file
+    /// </summary>
+    public class PackageHasSignatureValidator : Validator
+    {
+        private const string NupkgSignatureFile = ".signature.p7s";
+
+        private readonly ILogger<PackageHasSignatureValidator> _logger;
+
+        public PackageHasSignatureValidator(
+            IDictionary<FeedType, SourceRepository> feedToSource,
+            ILogger<PackageHasSignatureValidator> logger)
+          : base(feedToSource, logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        protected override async Task<bool> ShouldRunAsync(ValidationContext data)
```

Comment:
Throughout the file rename `data` to `context`.

Reply:
Done